src/v1alpha3/echo_client.ts

```diff
@@ -0,0 +1,233 @@
+    // Ensure that options include the service address and port.
+    opts = Object.assign(
+        {
+          clientConfig: {},
+          port: (this.constructor as typeof EchoClient).port,
+          servicePath: (this.constructor as typeof EchoClient).servicePath,
+        },
+        opts);
+
+    // Create a `gaxGrpc` object, with any grpc-specific options
+    // sent to the client.
+    opts.scopes = (this.constructor as typeof EchoClient).scopes;
+    const gaxGrpc = new gax.GrpcClient(opts);
+
+    // Save the auth object to the client, for use by other methods.
+    this.auth = gaxGrpc.auth;
+
+    // Determine the client header string.
+    const clientHeader = [
+      `gl-node/${process.version}`,
+      `grpc/${gaxGrpc.grpcVersion}`,
+      `gax/${gax.version}`,
+      `gapic/${version}`,
+    ];
+    if (opts.libName && opts.libVersion) {
+      clientHeader.push(`${opts.libName}/${opts.libVersion}`);
+    }
+
+    const gaxProtos = Object.assign(
+        {},
+        gaxGrpc.loadProto(
+            path.join(__dirname, '..', '..', 'protos'),
+            'google/showcase/v1alpha3/echo.proto'));
+
+    // Some of the methods on this service return "paged" results,
+    // (e.g. 50 results at a time, with tokens to get subsequent
+    // pages). Denote the keys used for pagination and results.
+    this._descriptors.page = {
+      pagedExpand:
+          new gax.PageDescriptor('pageToken', 'nextPageToken', 'responses')
+    };
+
+    // Some of the methods on this service provide streaming responses.
+    // Provide descriptors for these.
+    this._descriptors.stream = {
+      expand: new gax.StreamDescriptor(gax.StreamType.SERVER_STREAMING),
+      collect: new gax.StreamDescriptor(gax.StreamType.CLIENT_STREAMING),
+      chat: new gax.StreamDescriptor(gax.StreamType.BIDI_STREAMING)
+    };
+
+    // Put together the default options sent with requests.
+    const defaults = gaxGrpc.constructSettings(
+        'google.showcase.v1alpha3.Echo', gapicConfig as gax.ClientConfig,
+        opts.clientConfig || {}, {'x-goog-api-client': clientHeader.join(' ')});
+
+    // Set up a dictionary of "inner API calls"; the core implementation
+    // of calling the API is handled in `google-gax`, with this code
+    // merely providing the destination and request information.
+    this._innerApiCalls = {};
+
+    // Put together the "service stub" for
+    // google.showcase.v1alpha2.Echo.
+    const echoStub =
+        // @ts-ignore We cannot check types that are loaded in runtime.
+        gaxGrpc.createStub(gaxProtos.google.showcase.v1alpha3.Echo, opts);
+
+    const echoStubMethods =
+        ['echo', 'expand', 'collect', 'chat', 'pagedExpand', 'wait'];
+
+    for (const methodName of echoStubMethods) {
+      this._innerApiCalls[methodName] = gax.createApiCall(
+          echoStub.then(
+              // @ts-ignore
+              stub => function() {
+                const args = Array.prototype.slice.call(arguments, 0);
```

Comment:
lets use `...args`, and drop the need to do the funky slice

Reply:
Fixed! Thank you for this suggestion!